parlai/core/params.py

```diff
@@ -673,12 +673,20 @@ def add_parlai_args(self, args=None):
             type=int,
             help='batch size for minibatch training schemes',
         )
+        parlai.add_argument(
+            '-nw',
+            '--numworkers',
+            type=int,
+            default=1,
+            help='how many workers the Pytorch dataloader should use',
+            hidden=True,
+        )
         parlai.add_argument(
             '-dynb',
             '--dynamic-batching',
             default=None,
             type='nonestr',
-            choices={'none', 'full', 'batchsort'},
+            choices={None, 'full', 'batchsort'},
```

Comment:
This seems off. Are choices processed after parsing with the type?

Reply:
yes that is exactly what was going on - if i specified `'none'`, it would tell me that `None` was not a valid choice:

```
Parse Error: argument -dynb/--dynamic-batching: invalid choice: None (choose from 'full', 'batchsort', 'none')
```